plugin/path_role.go

```diff
@@ -318,20 +427,34 @@ func (b *GcpAuthBackend) role(s logical.Storage, name string) (*gcpRole, error)
 }
 
 // storeRole saves the gcpRole to storage.
+// The returned response may contain either warnings or an error response,
+// but will be nil if error is not nil
 func (b *GcpAuthBackend) storeRole(s logical.Storage, roleName string, role *gcpRole) (*logical.Response, error) {
-	if err := role.validate(b.System()); err != nil {
+	var resp *logical.Response
+	warnings, err := role.validate(b.System())
+
+	if err != nil {
 		return logical.ErrorResponse(err.Error()), nil
+	} else if len(warnings) > 0 {
+		resp = &logical.Response{
+			Warnings: warnings,
+		}
 	}
 
 	entry, err := logical.StorageEntryJSON(fmt.Sprintf("role/%s", roleName), role)
 	if err != nil {
 		return nil, err
 	}
 
-	return nil, s.Put(entry)
+	if err := s.Put(entry); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
 }
 
 type gcpRole struct {
+	// TODO(emilyyme): Make this consistent with input arg name ("type").
```

Comment:
It's not super important though, and be careful about backwards compatibility if you want to change it. e.g. it's fine to change the name of the actual struct member but changing the struct tags can lead to danger without an upgrade path.

Reply:
Yeah, pretty much why I put a TODO - I wasn't really sure the best way to migrate it. I can remove it if you think it's not worth the effort.